packages/storybook/src/Accordion/Accordion.stories.tsx

```diff
@@ -189,3 +182,209 @@ export function FullyControlled() {
     </Accordion>
   )
 }
+
+export function AccordionHookSingle() {
+  const {
+    setExpandedItems,
+    toggleExpanded,
+    checkIsExpanded,
+    register,
+    getHeaderProps,
+  } = useAccordion({
+    onChange: ({ expandedItems }) => console.info({ expandedItems }),
+    defaultExpandedItems: ['second'],
+    allowMultiple: false,
+  })
+
+  return (
+    <Flex sx={{ flexDirection: 'column', p: 6 }}>
+      <Accordion>
+        <AccordionItem>
+          <AccordionHeader
+            tabIndex={-1}
+            ref={node =>
+              register(node, { key: 'first', role: 'accordionHeader' })
```

Comment:
@alexanderchan is there a way that we don't have to require the developer to pass the `role` option? I know you are using this to determine which method to call when registering, but ideally accessibility attributes should be handled in the hook, IMO.

Reply:
One way could be to render a `header` element for the Accordion header so that you can check `ref.tagName === 'HEADER'` to determine which ref is the header ref.

Reply:
ahh nice was wondering if we could do that without passing the role through.  I'll set the body to `section` too.